docs/DeveloperGuide.md

```diff
@@ -2,36 +2,524 @@
+
+4. Command Creation: The Parser creates an instance of the appropriate Command class, passing along any required arguments.
+This encapsulates the user's request in an executable command object.
+
+5. Error Handling: If the user input does not match any recognized command or has formatting errors, the Parser generates
+an "IncorrectCommand" with an error message, providing feedback to the user.
+
+By structuring the parsing process this way, the application ensures that user input is correctly interpreted and
+translated into executable commands for the subsequent phases of the application.
+
+The parser class uses the below method to achieve its functionality
+
+- `parseCommand(String userInput)` - This method takes the user's input as a parameter and processes it to identify the
+  command keyword and any associated arguments. It then recognizes the specific command and prepares it for execution.
+
+Given below is an example of how the login system class works.
+
+Suppose a user enters the command `add /n Paracetamol /d 2023-12-31 /s ABC123 /q 100`. The Parser first splits this input
+into the command word "add" and the arguments. It then recognizes the "add" command, extracts the drug details (name,
+expiry date, serial number, and quantity), and creates an instance of the "`AddCommand`" with these details. If the user
+enters an invalid command or incorrect formatting, the Parser provides feedback to guide the user, ensuring a seamless
+interaction between the user and the application.
+
+The Parser is a crucial component that forms the bridge between user intentions and the core functionality of the stock
+management system.
+
+The following sequence diagram shows how the parser class works when the program is running.
+<img src="UML Diagrams/ParserDiagram.png" width="500">
+
+---
+
+### Main data structures
+
+---
+
+#### Implementation
+
+##### Drug
+
+The Drug class is very basic class for now. It only contains the product name as well as it's expiry date, but we will
+soon add some new properties such as product description, etc...
+
+##### Inventory
+
+The Inventory class is used to keep track of the quantity of product in stock. The hash map seemed to be the most
+appropriate data structure to match a product id to a quantity and a product entity which are encapsulated in a "
+StockEntry" class.
+
+##### Cart
+
+The Cart class is used to represent an ongoing transaction : to perform a sale, the vendor can add different products
+with their respective quantities in a cart which will be deducted from the inventory at the checkout.
+To represent this, we chose to use an arraylist of "CartEntry" classes which reprensent a product/quantity tuple.
+
+##### SalesList
+
+The SalesList is used to represent every past sales in order to create some statistics and reports. This class is only a
+list of subclasses representing validated carts.
+
+---
+
+## Commands
+
```

Comment:
For commands, adding sequence diagrams would help visualise and understand the steps of each command